### PR TITLE
Add support for Iterable as interface parameter type

### PR DESCRIPTION
**Pull request by verhasi**
Sometimes it is handy to give values as an Iterable to the backing bean. Also, the validation of the Iterable as notNull and its elements should be added.

### CODE REVIEW

File: fluapigen-processor/src/main/java/io/toolisticon/fluapigen/processor/ModelInterfaceMethodParameter.java [hunk 1]
```diff
@@ -218,6 +224,10 @@ else if (backBeanField.get().isArray()) {
         return "";
     }
 
+    private String generateSourceForIterableToList(String parameterName){
+        return "StreamSupport.stream(((Iterable<Class<? extends Processor>>)" + parameterName + ").spliterator(), false).collect(Collectors.toList())";
```
Comment (tobiasstamann) [inline fluapigen-processor/src/main/java/io/toolisticon/fluapigen/processor/ModelInterfaceMethodParameter.java:228]: We might need to change this to also support Sets as backing bean fields

**Comment by tobiasstamann** [comment]
Need to wait until APTK version is available is maven central to update version

File: fluapigen-processor/src/main/java/io/toolisticon/fluapigen/processor/ModelInterfaceMethodParameter.java [hunk 2]
```diff
@@ -218,6 +224,10 @@ else if (backBeanField.get().isArray()) {
         return "";
     }
 
+    private String generateSourceForIterableToList(String parameterName){
```
Comment (verhasi) [inline fluapigen-processor/src/main/java/io/toolisticon/fluapigen/processor/ModelInterfaceMethodParameter.java:227]: It is strange that I have got the notification email upon your comment but I can not find it here. Sure it is my bad. Anyway, even if we return a List it is a Collection that can be used both for addAll and (now I have fixed) for new backed bean instance creation.

**Comment by tobiasstamann** [comment]
I have removed the comment, i was wrong about it. The method returns a Collection that is added to the backing bean field via addAll so both List and Set will be supported.

**Comment by verhasi** [comment]
Ok, that is reasonable. The instance creation was missing indeed. Actually, this part of the tool is not "fully" tested yet AFAIS.

**Comment by tobiasstamann** [comment]
Sorry for the inconvenience.  Hope that the code of the library is somewhat comprehensible. It was really hard to create a working solution and i am still struggling to find the best architecture for building annotation processors.

**Comment by tobiasstamann** [comment]
I will check the existing tests and see what is missing. But some parts are really hard to test sufficiently. Especially when it comes to Generics or Inheritence.

**Comment by tobiasstamann** [comment]
APTK release is now available, i will check test coverage now.

**Comment by tobiasstamann** [comment]
Hi @verhasi,

there were some smaller issues that i have fixed now.
I have added some more integration tests for Sets and Iterables as well.

It's looking good so far, but i will check if more tests are needed.
So i guess that I will be able to trigger the release later today and take care for the changes in CUTE tomorrow.

File: fluapigen-processor/src/main/java/io/toolisticon/fluapigen/processor/ModelInterfaceMethodParameter.java (hunk 2 above)
Comment (tobiasstamann) [reply fluapigen-processor/src/main/java/io/toolisticon/fluapigen/processor/ModelInterfaceMethodParameter.java:227]: I had to remove the cast here, the parameter will always be of type Iterable including the correct Generic type declaration

Comment (verhasi) [reply fluapigen-processor/src/main/java/io/toolisticon/fluapigen/processor/ModelInterfaceMethodParameter.java:227]: Sure it was a mistake from my side. This specific cast is applicable only for the CUTE usage but Fluent API Generator is a generic tool. Thanks for catching.

**Comment by tobiasstamann** [comment]
Unfortunately there is another issue with the NotNull validator that need to be adressed before I can merge the change.
The NotNull validator is currently the only validator that should supports generic types, but can't since it is statically linked to Object.
Due to this  fact the resolving of correct method seems to be kind of broken.
I have to add support for generic validators first. 
I hope that i will be able to merge the change tomorrow.

**Comment by verhasi** [comment]
I am not depending on the next version release. I can go ahead with the locally built SNAPSHOT version without any issues. At first quality, then release. Sorry, if I caused additional work.

**Comment by tobiasstamann** [comment]
No, need to excuse. I already had some doubts regarding the validators. It's way cleaner if it supports Collections/Iterables as well. 
I was especially unsure regarding ambiguous Validators like MinLength/MaxLength which can target both Strings as single value or arrays/Collections. 
And of course i am very glad to receive any kind feedback i can get since I do this projects mainly in solo mode :)

**Pull request merged**